gpu/octree/src/octree.cpp

```diff
@@ -170,6 +170,12 @@ void pcl::gpu::Octree::approxNearestSearch(const Queries& queries, NeighborIndic
     static_cast<OctreeImpl*>(impl)->approxNearestSearch(q, results);
 }
 
+void pcl::gpu::Octree::nearestKSearchBatch(const Queries& queries, int k, NeighborIndices& results) const
+{
+    ResultSqrDists sqr_distances(queries.size() * k);
```

Comment:
You should not force users to have to preallocate an out parameter. The method should do that automatically.

Reply:
No, there wasn't strong reason for it, I will change it. I was simply looking at the existing test code, where the other out parameter is also preallocated, and using the same format.
```cpp
    pcl::gpu::NeighborIndices result_device(data.tests_num, k);    
```

I only realized now that eventhough it's pre-allocated there, its actually re-allocated inside the method, so the pre allocation isn't really necessary anyway.
```cpp
    results.create(static_cast<int> (queries.size()), k);	    
```